saltproc/tests/test_simulation.py

```diff
@@ -1,41 +1,440 @@
+
+    # close the file
+    db.close()
+
+    # delete test file
+    os.remove(db_file)
+
+    # use original db path
+    simulation.db_path = db_path_old
+
+
+def test_store_run_step_info():
+    """
+    This unit test checks that the entries ``store_run_step_info()`
+    stores in the database match the corresponding entries from the input
+    explicity in value and implicitly in type.
+    """
+    # read data
+    simulation.sim_depcode.read_depcode_step_param()
+    step_info = simulation.sim_depcode.param
+
+    # we want to keep the old path for other sims, but for this
+    # test we'll want a fresh db
+    db_path_old = simulation.db_path
+    db_file = serpent.iter_inputfile + '.h5'
+    simulation.db_path = db_file
+
+    # store data at
+    simulation.store_run_step_info()
+
+    # read stored data
+    try:
+        db = tb.open_file(simulation.db_path, mode='r',
+                          filters=simulation.compression_params)
+    except Exception:
+        print('Unable to assign correct value to db.\
+              See error stack for more info.')
+
+    tstep_info = db.root.simulation_parameters[0]
+    try:
+        assert np.array_equal(tstep_info[0],
+                              step_info['beta_eff'].astype('float32'))
+        assert np.array_equal(tstep_info[1],
+                              step_info['breeding_ratio'].astype('float32'))
+        assert tstep_info[2] == simulation.burn_time
+        assert np.array_equal(tstep_info[3],
+                              step_info['delayed_neutrons_lambda'].
+                              astype('float32'))
+        assert tstep_info[4] == step_info['fission_mass_bds'].astype('float32')
+        assert tstep_info[5] == step_info['fission_mass_eds'].astype('float32')
+        assert np.array_equal(tstep_info[6],
+                              step_info['keff_bds'].astype('float32'))
+        assert np.array_equal(tstep_info[7],
+                              step_info['keff_eds'].astype('float32'))
+        assert tstep_info[8] == step_info['memory_usage'].astype('float32')
+        assert tstep_info[9] == step_info['power_level'].astype('float32')
+        assert tstep_info[10] == step_info['execution_time'].astype('float32')
+    except AssertionError:
+        db.close()
+        os.remove(db_file)
+        simulation.db_path = db_path_old
+        raise AssertionError("incorrect Value")
+    except KeyError:
+        db.close()
+        os.remove(db_file)
+        simulation.db_path = db_path_old
+        raise KeyError("incorrect key")
+    except BaseException:
+        db.close()
+        os.remove(db_file)
+        simulation.db_path = db_path_old
+        print("something went wrong. See error stack for details")
+
+    # close the file
+    db.close()
+
+    # delete test file
+    os.remove(db_file)
+
+    # use original db path
+    simulation.db_path = db_path_old
+
+
+def test_read_k_eds_delta():
+    assert simulation.read_k_eds_delta(7) is False
```

Comment:
Why does simulation.read_k_eds_delta only come back as false?

Reply:
We should definitely have more checks in this test. I'll work on it.